azure/packages/azure-client/src/AzureClient.ts

```diff
@@ -303,5 +305,16 @@ export class AzureClient {
 		fluidContainer.attach = attach;
 		return fluidContainer;
 	}
+
+	private async getContainerEntryPoint(container: IContainer): Promise<IRootDataObject> {
+		const rootDataObject: FluidObject<IRootDataObject> | undefined =
+			await container.getEntryPoint();
+		assert(rootDataObject !== undefined, "entryPoint must exist");
+		// ! This "if" is needed for back-compat (older instances of IRootDataObject may not have the IRootDataObject property)
```

Comment:
When can we remove this back-compat?  If we know a priori it's nice to put it in the comment to give confidence when removing it.

Reply:
I think it can be removed in `next` once the `IRootDataObject` property is required. Might be worth making a separate task to track this as well as improving some of the typing around entryPoint

Reply:
[AB#6341](https://dev.azure.com/fluidframework/internal/_workitems/edit/6341)